test/queue.f90

```diff
@@ -0,0 +1,171 @@
+! queues.f90 --
```

Comment:
Two items: 1) indicate that it taken from wherever and adapted/changes, and 2) simplify this (e.g., see comments on specific lines)

Reply:
Just to let you know, the source I adapted the code from uses the same license as polyMPO.